[rocprofiler-compute] Roofline runtime compilation patch (#2232)
ROCm/rocm-systems: projects/rocprofiler-compute/CHANGELOG.md

    * kernel: Counters are collected in a round robin fashion for unique kernels.
    * kernel_launch_params: Counters are collected in a round robin fashion for unique kernels having the exact same launch parameters.

> * Runtime Compilation of Roofline benchmarking

Reviewer: ```suggestion
* Runtime compilation of Roofline benchmarking:
```

>   * GPU kernels from [rocm-amdgpu-bench](https://github.com/ROCm/rocm-amdgpu-bench) repository are moved into the Rocprofiler-Compute project and are compiled at runtime using local HIP and HIPRTC python wrappers.

Reviewer: ```suggestion
  * GPU kernels from [rocm-amdgpu-bench](https://github.com/ROCm/rocm-amdgpu-bench) repository are moved into the ROCm Compute Profiler and are compiled at runtime using local HIP and HIPRTC Python wrappers.
```

>   * Roofline binaries compiled from rocm-amdgpu-bench repository have been removed from the project, as Roofline Runtime Compilation performs the same work as the Roofline binaries.

Reviewer: ```suggestion
  * Roofline binaries compiled from [rocm-amdgpu-bench](https://github.com/ROCm/rocm-amdgpu-bench) repository have been removed from the project, as Roofline runtime compilation performs the same work as the Roofline binaries.
```

>   * User can collect standalone Roofline empirical peaks without running entire Rocprofiler-Compute's profile mode through an entry point in [benchmark.py](https://github.com/ROCm/rocm-systems/blob/HEAD/projects/rocprofiler-compute/src/utils/benchmark.py) - running `benchmark.py` replaces calling standalone Roofline binary.

Reviewer: ```suggestion
  * You can collect standalone Roofline empirical peaks without running the entire ROCm Compute Profiler's profile mode, through an entry point in [benchmark.py](https://github.com/ROCm/rocm-systems/blob/HEAD/projects/rocprofiler-compute/src/utils/benchmark.py). Running the `benchmark.py` Python file replaces calling standalone Roofline binary.
```

>   * User can collect standalone Roofline empirical peaks without running entire Rocprofiler-Compute's profile mode through an entry point in [benchmark.py](https://github.com/ROCm/rocm-systems/blob/develop/projects/rocprofiler-compute/src/utils/benchmark.py) - running `benchmark.py` replaces calling standalone Roofline binary.

Reviewer 2: Good bot.
